api/src/main/java/com/datatorrent/api/ClusterMode.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.api;
+
+import org.apache.hadoop.conf.Configuration;
+
+/**
+ *
+ */
+public abstract class ClusterMode<H extends Launcher.AppHandle> extends Launcher<H>
+{
+
+  public static final Attribute<String> LIB_JARS = new Attribute<String>(new StringCodec.String2String());
+  public static final Attribute<String> ORIGINAL_APP_ID = new Attribute<String>(new StringCodec.String2String());
+  public static final Attribute<String> QUEUE_NAME = new Attribute<String>(new StringCodec.String2String());
```

Comment:
@PramodSSImmaneni : I have following feedback on this
1. Attribute is used to define variables that control Platform behavior. These defined here don't control platform behavior
2. When user launches app using cli, user can pass these as command line options but they are not set as attributes but now with cluster mode implementation you are asking user to set them as attributes. There is discrepancy here.
3. Using cli, user can also set more options which is not reflected in this.